src/jit/assertionprop.cpp

```diff
@@ -3768,8 +3768,8 @@ EXPSET_TP Compiler::optImpliedByConstAssertion(AssertionDsc* constAssertion)
 
         case O2K_CONST_INT:
             // Is the const assertion's constant equal/not equal to the implied assertion?
-            usable = ((impAssertion->assertionKind == OAK_EQUAL) && (impAssertion->op2.u1.iconVal == iconVal)) ||
-                      (impAssertion->assertionKind == OAK_NOT_EQUAL) && (impAssertion->op2.u1.iconVal != iconVal);
+            usable = (((impAssertion->assertionKind == OAK_EQUAL) && (impAssertion->op2.u1.iconVal == iconVal))) ||
```

Comment:
Thanks for spotting it. It seems I have added the parens above accidentally together with adding the ones below. The ones below are needed, the ones above are useless.